darts/models/forecasting/forecasting_model.py

```diff
@@ -1430,9 +1456,22 @@ def fit(self, series: TimeSeries, future_covariates: Optional[TimeSeries] = None
                 series.has_same_time_as(future_covariates),
                 "The provided `future_covariates` series must contain at least the same time steps/"
                 "indices as the target `series`.",
-                logger,
+                logger=logger,
+            )
+            self._expect_future_covariates = True
+
+        self.encoders = self.initialize_encoders()
+        if self.encoders.encoding_available:
+            _, future_covariates = self.generate_fit_encodings(
+                series=series,
+                past_covariates=None,
+                future_covariates=future_covariates,
+            )
+            raise_if_not(
```

Comment:
Do we have to keep this? The check `has_same_time_as()` may take a small performance toll.